### PR TITLE
CORE-13551 Merge branch release/os/5.1

### PR DESCRIPTION
**Pull request by YashNabar**
Sync re-registration feature branch with release/os/5.1
https://github.com/corda/corda-api/pull/1110
https://github.com/corda/corda-e2e-tests/pull/105
https://github.com/corda/corda-cli-plugin-host/pull/166

Conflicts:
- charts/corda/Chart.yaml
- components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MGMResourceClientImpl.kt
- components/membership/membership-client-impl/src/test/kotlin/net/corda/membership/impl/client/MGMResourceClientTest.kt
- components/membership/membership-p2p-impl/src/integrationTest/kotlin/net/corda/membership/impl/p2p/MembershipP2PIntegrationTest.kt
- components/membership/membership-p2p-impl/src/main/kotlin/net/corda/membership/impl/p2p/handler/RegistrationRequestHandler.kt
- components/membership/membership-p2p-impl/src/test/kotlin/net/corda/membership/impl/p2p/MembershipP2PProcessorTest.kt
- components/membership/membership-persistence-service-impl/src/main/kotlin/net/corda/membership/impl/persistence/service/handler/AddNotaryToGroupParametersHandler.kt
- components/membership/membership-persistence-service-impl/src/main/kotlin/net/corda/membership/impl/persistence/service/handler/RegistrationStatusHelper.kt
- components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/AddNotaryToGroupParametersHandlerTest.kt
- components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/ApproveRegistrationHandler.kt
- components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/DeclineRegistrationHandler.kt
- components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/StartRegistrationHandler.kt
- components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/member/DynamicMemberRegistrationService.kt
- components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/mgm/MGMRegistrationService.kt
- components/membership/registration-impl/src/test/kotlin/net/corda/membership/impl/registration/dynamic/RegistrationProcessorTest.kt
- components/membership/registration-impl/src/test/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/DeclineRegistrationHandlerTest.kt
- components/membership/registration-impl/src/test/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/StartRegistrationHandlerTest.kt
- components/membership/registration-impl/src/test/kotlin/net/corda/membership/impl/registration/dynamic/member/DynamicMemberRegistrationServiceTest.kt
- gradle.properties
- processors/rest-processor/src/integrationTest/resources/swaggerBaseline.json
- testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/ClusterBuilder.kt
- testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MGMUtils.kt
- testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/MembershipUtils.kt
- testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/Utils.kt
- testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/Member.kt

### CODE REVIEW

File: components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/mgm/MGMRegistrationService.kt [hunk 1]
```diff
@@ -211,9 +182,9 @@ class MGMRegistrationService @Activate constructor(
                 val groupParameters = groupParametersPersistenceResult.getOrThrow()
                 groupParametersWriterService.put(member, groupParameters)
 
-                mgmRegistrationOutputPublisher.publish(mgmInfo.memberInfo)
-
                 expirationProcessor.scheduleProcessingOfExpiredRequests(member)
```
Comment (nikinagy) [inline components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/mgm/MGMRegistrationService.kt:185]: I would like to keep this as the last step. The record publishing should come first to make sure our info got published successfully.

Comment (YashNabar) [reply components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/mgm/MGMRegistrationService.kt:185]: I think it expects those records to be returned for the retrying logic. @yift-r3 what do you think?

Comment (yift-r3) [reply components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/mgm/MGMRegistrationService.kt:185]: I think that if we return the records, the processor will publish them so we don't need to worry if they have been published successfully.

**Review by williamvigorr3 — DISMISSED** [review]
LGTM - I checked everything that had conflicts (and any additional changes to auto merged files).

**Review by nikinagy — APPROVED** [review]
LGTM

**Pull request merged**